### ISSUE TITLE
"$this" is not allowed in enum methods, but it should be

### ISSUE THREAD

**Issue by mhgtlc** [squizlabs/PHP_CodeSniffer#3827]
**Describe the bug**

**Code sample**
```php
enum Status
{
    case DRAFT;
    case PUBLISHED;
    case ARCHIVED;
    
    public function color(): string
    {
        return match($this) 
        {
            Status::DRAFT => 'grey',   
            Status::PUBLISHED => 'green',   
            Status::ARCHIVED => 'red',   
        };
    }
}
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset name="PHP_CodeSniffer">
    ...
    <config name="testVersion" value="8.1-"/>
    <rule ref="PHPCompatibility"/>
    ...
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
  10 | ERROR | "$this" can no longer be used in a plain function or
     |       | method since PHP 7.1.
----------------------------------------------------------------------
```

**Expected behavior**
$this should be allowed inside enum methods, please take a look at this example: https://stitcher.io/blog/php-enums

**Versions (please complete the following information):**

 - PHP: 8.1
 - PHPCS: 3.7.2
 - Standard: PSR12

**Comment by jrfnl** [comment]
Closing as reported to the wrong repo.

> Please report bugs for externally maintained sniffs to the appropriate external standard repository (not here).

P.S.: I happen to know this has already been fixed in the PHPCompatibility `develop` branch.

**Comment by golliholzland** [comment]
@jrfnl could you include the commit or version to use, in order to fix this

**Comment by jrfnl** [comment]
@golliholzland As stated before: this question is about a sniff from another - externally maintained - project. Check the repo of that project and don't ask here.

**Comment by DanielRuf** [comment]
Hm, I'm only using squizlabs/php_codesniffer 3.13.5 (which seems to be https://github.com/PHPCSStandards/PHP_CodeSniffer/releases/tag/3.13.5 from https://github.com/PHPCSStandards/PHP_CodeSniffer.git) and get this for an enum like shown in the issue description.

> this question is about a sniff from another - externally maintained - project

How can I find this out?
I guess the following is meant:

`phpcs --standard=config/phpcs_phpcompatibility.xml --runtime-set testVersion 8.1-8.4 --colors -p`

And the xml file looks like this:
```
<?xml version="1.0"?>
<ruleset name="Custom PHP Compatibility" namespace="Planner-Suite\CS\Compatibility">
    <exclude-pattern>*/js/*</exclude-pattern>
    <exclude-pattern>*/lib/*</exclude-pattern>
    <exclude-pattern>*/node_modules/*</exclude-pattern>
    <exclude-pattern>*/public/*</exclude-pattern>
    <exclude-pattern>*/tests/*</exclude-pattern>
    <exclude-pattern>*/vendor/*</exclude-pattern>
    <exclude-pattern>*/vue-js/*</exclude-pattern>
    <config name="installed_paths" value="vendor/phpcompatibility/php-compatibility" />
    <rule ref="PHPCompatibility" />
</ruleset>
```

`phpcompatibility/php-compatibility` is at 9.3.5, which is https://github.com/PHPCompatibility/PHPCompatibility/releases/tag/9.3.5

But there are no new stable versions, only alpha releases:
https://github.com/PHPCompatibility/PHPCompatibility/tags

Does that mean I have to switch to https://github.com/PHPCompatibility/PHPCompatibility/releases/tag/10.0.0-alpha2 instead of 9.3.5?

**Comment by jrfnl** [comment]
Yes.

Locking this issue as clearly people do not (want to) understand that asking about an package which is not maintained in this repo is just wasting maintainer time. If you just want to dump your questions somewhere and hope someone answers, use StackOverflow.
